src/main/view/screen/IntroPanel.java

```diff
@@ -0,0 +1,87 @@
+package view.screen;
+
+import java.awt.Point;
+
+import javafx.beans.binding.Bindings;
+import javafx.event.EventHandler;
+import javafx.scene.Group;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.StackPane;
+import javafx.scene.media.Media;
+import javafx.scene.media.MediaPlayer;
+import javafx.scene.media.MediaView;
+import view.GameModelAdapter;
+import view.Panel;
+import view.PanelManager;
+import view.ViewEnum;
+import view.assets.AssetManager;
+
+public class IntroPanel extends Panel {
+	private static final int ASPECT_RATIO_X = 16;
+	private static final int ASPECT_RATIO_Y = 9;
+	PanelManager panelManager;
+	Group root;
+	Media intro = getAssets().getIntro();
+	StackPane video = new StackPane();
+	MediaPlayer player = new MediaPlayer(intro);
+	MediaView view = new MediaView(player);
+	private boolean introStarted = false;
+
+	public IntroPanel(GameModelAdapter gameModelAdapter, Group root, PanelManager panelManager,
+			AssetManager assets, ViewEnum viewEnum) {
+		super(gameModelAdapter, assets, viewEnum);
+		this.root = root;
+		checkSkipped();
+		this.panelManager = panelManager;
+		view.fitWidthProperty().bind(Bindings.selectDouble(view.sceneProperty(), "width"));
+		view.fitHeightProperty().bind(Bindings.selectDouble(view.sceneProperty(), "height"));
+		view.setPreserveRatio(true);
+		video.getChildren().add(view);
+	}
+
+	private void checkSkipped() {
+		video.setOnMouseClicked(new EventHandler<MouseEvent>() { // Click to skip
+			public void handle(MouseEvent event) {
+				if (player.getStatus() == MediaPlayer.Status.PLAYING) {
+					player.stop();
+					panelManager.setMode(ViewEnum.MAIN_GAME);
+				}
+			}
+		});
+
+	}
+
+	public void draw(GraphicsContext g, Point screenDimensions, long currentPulse) {
+		//Centers Video On Screen
+		if ((double)screenDimensions.y/(double)screenDimensions.x < 
+				(double)ASPECT_RATIO_Y/(double)ASPECT_RATIO_X) {
+			view.setTranslateX((screenDimensions.x - ((ASPECT_RATIO_X * screenDimensions.y)/ASPECT_RATIO_Y))/2);
+		}
+		else {
+			view.setTranslateY((screenDimensions.y - ((ASPECT_RATIO_Y * screenDimensions.x)/ASPECT_RATIO_X))/2); 
+		}
+		if (!introStarted) {
+			playIntroVideo();
+			introStarted = true;
+		} else {
+			if (player.getCurrentTime().toMillis() == 39894) { //Last Milli of Video
```

Comment:
slightly confusing with the name of "player" since there's a player class, but you can leave it if you don't feel like changing it